lib/assets/Asset.js

```diff
@@ -571,6 +571,17 @@ class Asset extends EventEmitter {
     return null;
   }
 
+  get baseUrl() {
+    if (this.isInline) {
+      const nonInlineAncestor = this.nonInlineAncestor;
+      if (nonInlineAncestor) {
+        return this.nonInlineAncestor.baseUrl;
```

Comment:
This is a pretty nice place to put it. It's certainly more intuitive to just access `asset.baseUrl` instead of remembering that in some cases it can be inline and then having to specifically look for `nonInlineAncestor`

Reply:
Yes, I agree!

Although we do need to keep `Relation#baseUrl` as well because `sourceRoot` in source maps only affects the `SourceMapSource` relation: https://github.com/assetgraph/assetgraph/blob/3f1b3d2602f69e120fa43ddb4d495e53da658e6f/lib/relations/SourceMap/SourceMapSource.js#L12-L22